cmd/main.go

```diff
@@ -30,6 +35,11 @@ func main() {
 
 	// start server
 	port := htsconfig.GetPort()
-	fmt.Printf("Server started on port %s!\n", port)
-	http.ListenAndServe(":"+port, nil)
+	if *NoTLS {
+		log.Infof("Insecure HTTP Server started on port %s!", port)
+		log.Fatal(http.ListenAndServe(":"+port, nil))
+	} else {
+		log.Infof("Server started on port %s!", port)
+		log.Fatal(http.ListenAndServeTLS(":"+port, *TLSCert, *TLSKey, nil))
+	}
```

Comment:
```suggestion
	server := &http.Server{
		Addr:              ":" + htsconfig.GetPort(),
		ReadTimeout:       30 * time.Second,
		WriteTimeout:      30 * time.Second,
		IdleTimeout:       30 * time.Second,
		ReadHeaderTimeout: 30 * time.Second,
	}

	if htsconfig.GetTLSCert() != "" && htsconfig.GetTLSKey() != "" {
		if err := server.ListenAndServeTLS(htsconfig.GetTLSCert(), htsconfig.GetTLSKey()); err != nil {
			panic(err)
		}
	} else {
		if err := server.ListenAndServe(); err != nil {
			panic(err)
		}
	}
```

Use the already existing configuration functionality instead of adding flag support.

Reply:
yeah, that's nicer! :) Not gonna use the config system though - it's big and messy, and we're looking for quick fixes while we decide if we're even gonna use this.

Reply:
Do it properly or all work will be mostly wasted.

Reply:
No. Doing things "properly" is wasted if we ditch this product. And if we adopt it we're gonna add the config system we use for the other services anyway...